pkg/patterns/declarative/reconciler.go

```diff
@@ -134,6 +136,40 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+
+	// dynamic config
+	dynamicClientset, err := dynamic.NewForConfig(r.config)
```

Comment:
We may want to construct this earlier and cache it in the Reconciler object; though it doesn't matter as much for the dynamic client which doesn't have as much state.